cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -0,0 +1,246 @@
+    if (foundDefaultBranch != null) {
+      return null;
+    }
+    return new RepositoryValidationFailure(String.format(
+      "Default branch not found in remote repository. Ensure ref '%s' already exists.",
+      defaultBranchRefName));
+  }
+
+  /**
+   * Validated the configuration provided to this manager.
+   *
+   * @return a {@link List} of {@link  RepositoryValidationFailure}s occurred during validation.
+   */
+  public RepositoryValidationFailure validateConfig() throws IOException {
+    // Validate existence of secret in Secure Store.
+    RepositoryConfig config = repositoryConfig;
+    LOG.info("Checking existence of secret in secure store: " + config.getAuth().getTokenName());
+    try {
+      store.get(namespaceId.getNamespace(), config.getAuth().getTokenName()).get();
+    } catch (Exception e) {
+      LOG.error("Failed to get auth credentials from secure store.", e);
+      return new RepositoryValidationFailure("Failed to get password/token from secure store.");
+    }
+
+    // Try fetching heads in the remote repository.
+    FileRepository localRepo;
+    deleteIfRequired(validationRepoDirectory);
+    try {
+      localRepo = new FileRepository(validationRepoDirectory);
+    } catch (IOException e) {
+      throw new IOException(String.format("Failed to create local git repository at path '%s'",
+                                          validationRepoDirectory.getAbsolutePath()), e);
+    }
+
+    LOG.info("Created local repo in directory for validation: " + localRepoDirectory.getAbsolutePath());
+    try (Git git = new Git(localRepo)) {
+      LsRemoteCommand cmd = git.lsRemote().setRemote(config.getLink()).setHeads(true).setTags(false);
+      cmd = getAuthenticatedCommandWithTimeout(cmd);
+      RepositoryValidationFailure failure = validateDefaultBranch(cmd.call());
+      if (failure != null) {
+        return failure;
+      }
+    } catch (TransportException e) {
+      LOG.error("Failed to list remotes in remote repository.", e);
+      return new RepositoryValidationFailure(
+        "Failed to connect with repository. Authentication credentials invalid or the repository may not exist.");
+    } catch (Exception e) {
+      LOG.error("Failed to list remotes in remote repository.", e);
+      return new RepositoryValidationFailure(e.getMessage());
+    }
+    return null;
+  }
+
+  // Validated whether git is initialized.
+  private void validateInitialized() {
+    if (git == null) {
+      throw new IllegalStateException("Initialize source control manager before performing operation.");
+    }
+  }
+
+  public Path getBasePath() {
+    if (repositoryConfig.getPathPrefix() == null) {
+      return localRepoDirectory.toPath();
+    }
+    return localRepoDirectory.toPath().resolve(repositoryConfig.getPathPrefix());
+  }
+
+  private void deleteIfRequired(File directory) throws IOException {
+    if (directory.exists()) {
```

Comment:
We should add this in Dirutils as this seems to be very common usecase i.e deleteDirectoryIfExists

Reply:
Will do it in a follow up PR as it will increase the scope of this PR that has already been reviewed multiple times.